tests/unit/unit-walker.cpp

```diff
@@ -38,11 +45,62 @@ class UnitWalkerTests : public ::testing::Test,
     bvsort = s->make_sort(BV, 4);
     funsort = s->make_sort(FUNCTION, SortVec{ bvsort, bvsort });
     arrsort = s->make_sort(ARRAY, bvsort, bvsort);
+    boolsort = s->make_sort(BOOL);
   }
   SmtSolver s;
-  Sort bvsort, funsort, arrsort;
+  Sort bvsort, funsort, arrsort, boolsort;
 };
 
+class FreshTreeWalker : public TreeWalker {
+  int b_iter {};
+
+  using TreeWalker::TreeWalker;
+
+//  FreshTreeWalker(const smt::SmtSolver & solver,
+  //               bool clear_cache,
+    //             smt::UnorderedTermPairMap * ext_cache = nullptr)
+      //: solver_(solver), clear_cache_(clear_cache), ext_cache_(ext_cache){};
+
+  TreeWalkerStepResult visit_term(smt::Term & formula, smt::Term & term, std::vector<int> & path) override{
+
+    Sort boolsort = solver_->make_sort(BOOL);
+
+    string var_name = string("b") + to_string(b_iter);
+    Term b = solver_->make_symbol(var_name, boolsort);
+    b_iter++;
+
+    Op op = term->get_op();
+    if (!op.is_null())
+    {
+      TermVec cached_children;
+      Term c;
+      pair <Term, vector<int>> occurrence;
+      for (auto t : term)
+      {
+       // occurrence.first = t;
+        //occurrence.second = path;
+        //query_cache(t, occurrence);
+        cached_children.push_back(t);
```

Comment:
these are not cached. Same for the default implementation, so please change the name in both plasces.